### ISSUE TITLE
Archived on CRAN

### ISSUE THREAD

**Issue by gdkrmr** [mpadge/osmplotr#59]
osmplotr is archived on CRAN. Is the package still maintained?

**Comment by mpadge** [comment]
Yes, its definitely still maintained, and will be resubmitted to CRAN soon. This issue will serve as a bit more of a push to ensure that happens soon. Thanks!

**Comment by maelle** [comment]
@mpadge note that rgeos has been archived on CRAN (osmplotr uses it)

**Comment by maelle** [comment]
@mpadge any plan to resubmit to CRAN?

**Comment by mpadge** [comment]
Yes, it will happen, and shouldn't be too difficult now that #60 has been addressed.

**Comment by maelle** [comment]
any update? :smile_cat:

**Comment by maelle** [comment]
@mpadge friendly reminder :smile_cat:

**Comment by maelle** [comment]
@mpadge friendly reminder 😸

**Comment by mpadge** [comment]
Thank you so much @maelle for the gentle friendliness of your reminders. They finally got through: I've just submitted a first attempt at re-cran-ification, so we'll see how that goes ...

**Comment by maelle** [comment]
Fingers crossed!

**Comment by mpadge** [comment]
> Thanks,
>
> Please add \value to .Rd files regarding exported methods and explain the functions results in the documentation. Please write about the structure of the output (class) and also what the output means. (If a function does not return a value, please document that too, e.g. \value{No return value, called for side effects} or similar)
For more details: [<https://contributor.r-project.org/cran-cookbook/docs_issues.html#missing-value-tags-in-.rd-files>](https://contributor.r-project.org/cran-cookbook/docs_issues.html#missing-value-tags-in-.rd-files)
Missing Rd-tags:
     print_osm_map.Rd: \value
>
> \dontrun{} should only be used if the example really cannot be executed (e.g. because of missing additional software, missing API keys, ...) by the user. That's why wrapping examples in \dontrun{} adds the comment ("# Not run:") as a warning for the user. Does not seem necessary. Please replace \dontrun with \donttest.
Please unwrap the examples if they are executable in < 5 sec, or replace dontrun{} with \donttest{}.
For more details: [<https://contributor.r-project.org/cran-cookbook/general_issues.html#structuring-of-examples>](https://contributor.r-project.org/cran-cookbook/general_issues.html#structuring-of-examples)
>
>
> You write information messages to the console that cannot be easily suppressed.
It is more R like to generate objects that can be used to extract the information a user is interested in, and then print() that object. Instead of print()/cat() rather use message()/warning() or if(verbose)cat(..) (or maybe stop()) if you really have to write text to the console. (except for print, summary, interactive functions)
-> R/extract-highways.R; R/make-osm-map.R
For more details: [<https://contributor.r-project.org/cran-cookbook/code_issues.html#using-printcat>](https://contributor.r-project.org/cran-cookbook/code_issues.html#using-printcat)
>
> Please make sure that you do not change the user's options, par or working directory. If you really have to do so within functions, please ensure with an *immediate* call of on.exit() that the settings are reset when the function is exited.
e.g.:
...
oldpar <- par(no.readonly = TRUE) # code line i
on.exit(par(oldpar)) # code line i + 1
...
par(mfrow=c(2,2)) # somewhere after
...
> e.g.: -> R/adjust-colours.R; R/colour-mat.R; R/connect-highways.R
> If you're not familiar with the function, please check ?on.exit. This function makes it possible to restore options before exiting a function even if the function breaks. Therefore it needs to be called immediately after the option change within a function.
For more details: [<https://contributor.r-project.org/cran-cookbook/code_issues.html#change-of-options-graphical-parameters-and-working-directory>](https://contributor.r-project.org/cran-cookbook/code_issues.html#change-of-options-graphical-parameters-and-working-directory)
>
> Please fix and resubmit.

**Comment by maelle** [comment]
ICYMI https://github.com/DavisVaughan/extrachecks

**Comment by maelle** [comment]
@mpadge any update? 😸

**Comment by mpadge** [comment]
As you see, it was coming along... and still will be, when I find time. Thanks once again for the friendly reminder 😄

**Comment by maelle** [comment]
@mpadge your regular friendly reminder 😸

**Comment by maelle** [comment]
@mpadge your regular friendly reminder 😸

**Comment by mpadge** [comment]
Wow, you're so good ❤️ I reckon this time I should actually be able to find some time soon ...

**Comment by maelle** [comment]
👋 @mpadge, is any help needed?

**Comment by mpadge** [comment]
Thank you so much @maelle for continuing to gently 👉 👉 👉 here for over 2 years. I've just resubmitted new version to CRAN. You'll see here if it makes it through, and we can finally close this issue.

**Comment by mpadge** [comment]
Email 22 July 2026:

> Thanks,
>
> we still see:
>
> Please make sure that you do not change the user's options, par or working directory. If you really have to do so within functions, please ensure with an *immediate* call of on.exit() that the settings are reset when the function is exited.
e.g.:
> ...
> oldpar <- par(no.readonly = TRUE) # code line i
> on.exit(par(oldpar)) # code line i + 1
> ...
> par(mfrow=c(2,2)) # somewhere after
> ...
>
> ...
> oldwd <- getwd() # code line i
> on.exit(setwd(oldwd)) # code line i+1
> ...
> setwd(...) # somewhere after
> ...
> e.g.:
> If you're not familiar with the function, please check ?on.exit. This function makes it possible to restore options before exiting a function even if the function breaks. Therefore it needs to be called immediately after the option change within a function.
For more details: [<https://contributor.r-project.org/cran-cookbook/code_issues.html#change-of-options-graphical-parameters-and-working-directory>](https://contributor.r-project.org/cran-cookbook/code_issues.html#change-of-options-graphical-parameters-and-working-directory)
-> R/adjust-colours.R; R/colour-mat.R; R/connect-highways.R
> 
> Please fix and resubmit.
